openmetadata-service/src/main/java/org/openmetadata/csv/CsvUtil.java

```diff
@@ -94,6 +100,87 @@ public static List<String> fieldToEntities(String field) {
     return field == null ? null : listOf(field.split(ENTITY_TYPE_SEPARATOR));
   }
 
+  public static List<String> fieldToInternalArray(String field) {
+    // Split a fieldValue that contains multiple elements of an array separated by
+    // INTERNAL_ARRAY_SEPARATOR
+    if (field == null || field.isBlank()) {
+      return Collections.emptyList();
+    }
+    return listOf(field.split(Pattern.quote(INTERNAL_ARRAY_SEPARATOR)));
+  }
+
+  /**
+   * Parses a field containing key-value pairs separated by semicolons, correctly handling quotes.
+   * Each key-value pair may also be enclosed in quotes, especially if it contains delimiter like (SEPARATOR , FIELD_SEPARATOR).
+   *
+   * Input Example:
+   * "key1:value1;key2:value2;\"key3:value;with;semicolon\""
+   * Output: ["key1:value1", "key2:value2", "key3:value;with;semicolon"]
+   *
+   * @param field The input string with key-value pairs.
+   * @return A list of key-value pairs, handling quotes and semicolons correctly.
+   */
+  public static List<String> fieldToExtensionStrings(String field) {
```

Comment:
can we remove these comments. 
can we add some mocks to test this part of code

Reply:
updated logic of fieldToExtensionStrings  to use CSVParser